lib/shoulda/matchers/active_record/association_matchers/model_reflection.rb

```diff
@@ -13,7 +13,11 @@ def initialize(reflection)
           end
 
           def associated_class
-            reflection.klass
+            if polymorphic?
+              subject
+            else
+              reflection.klass
+            end
```

Comment:
I'll wait to see what happens when running all the specs, but when the relation is polymorphic,  the `reflection.klass` method raises an error. I found this while trying to create a spec to replicate a case where you have a polymorphic association or delegated_type one, but you also pass the `scopes` qualifier to that association. When trying to make a spec for that cause, I noticed this problem.

 My solution was to return a valid ActiveRecord object(which in that case, I returned the subject itself) because this will be used only to grab the value of the `where_values_hash` in the `ModelReflector#extract_relation_clause_from`, which I think doesn't matter which model you're using to. But we can think on another solution here.